airflow/jobs/scheduler_job.py

```diff
@@ -369,6 +369,9 @@ def _executable_task_instances_to_queued(self, max_tis: int, session: Session =
                     starved_pools.add(pool_name)
                     continue
 
+                # Make sure to emit metrics if pool has no starving tasks
+                pool_num_starving_tasks[pool_name] += 0
```

Comment:
The `+= 0` part looks so much like a bug, we should do something to make it less so.